packages/@angular/cli/utilities/validate-project-name.ts

```diff
@@ -2,30 +2,33 @@ import {oneLine, stripIndent} from 'common-tags';
 
 const SilentError = require('silent-error');
 
-const projectNameRegexp = /^[a-zA-Z][.0-9a-zA-Z]*(-[.0-9a-zA-Z]*)*$/;
+const projectNameRegexp = /^[a-zA-Z][0-9a-zA-Z]*([-. ][a-zA-Z][0-9a-zA-Z]*)*$/;
```

Comment:
We will never allow projects with:
1. starting dots (invalid npm package name),
1. multiple dots after each others (might not be valid file names on some platforms),
1. invalid URL characters (invalid npm package name), such as spaces.